features/partnerhistorie.feature

```diff
@@ -4,137 +4,259 @@ Functionaliteit: Tonen van Partnerhistorie
+Rule: Wanneer de sluiting/aangaan geheel onbekend is, wordt voor de filtering aangenomen dat de persoon deze partner altijd heeft gehad.
 
-  Scenario: datumTotEnMet gelijk aan datum aangaan
-    Als de partnerhistorie wordt gevraagd van de persoon met burgerservicenummer 999991553 met datumVan "2009-10-02" en datumTotEnMet "2009-11-02"
-    Dan worden de huwelijken/partners teruggegeven in de volgorde en met waarden:
-      | # | naam.voornamen | aangaanHuwelijkPartnerschap.datum | ontbindingHuwelijkPartnerschap.datum |
-      | 0 | Karel          | 2009-11-02                        |                                      |
+  @gba
+  Scenario: Raadpleeg op peildatum,  onbekende datum aangaan huwelijk/partnerschap
+    Als de partnerhistorie wordt geraadpleegd met de volgende parameters
+    | naam                | waarde                |
+    | type                | RaadpleegMetPeildatum |
+    | burgerservicenummer | 999991418             |
+    | fields              | naam.voornamen, aangaanHuwelijkPartnerschap.datum, ontbindingHuwelijkPartnerschap.datum |
+    | peildatum           | 1925-01-01            |
+    Dan worden de (ex-)partner-gegevens geretourneerd en met waarden:
+    | naam.voornamen | aangaanHuwelijkPartnerschap.datum | ontbindingHuwelijkPartnerschap.datum |
+    | Annechien      | 00000000                          |                                      |
 
-  Scenario: gedeeltelijk onbekende datum huwelijkssluiting
-    Als de partnerhistorie wordt gevraagd van de persoon met burgerservicenummer 999991553 met peildatum "1936-01-01"
+  @gba
+  Scenario: Raadpleeg op periode,  onbekende datum aangaan huwelijk/partnerschap
+    Als de partnerhistorie wordt geraadpleegd met de volgende parameters
+    | naam                | waarde                |
+    | type                | RaadpleegMetPeriode   |
+    | burgerservicenummer | 999991418             |
+    | fields              | naam.voornamen, aangaanHuwelijkPartnerschap.datum, ontbindingHuwelijkPartnerschap.datum |
+    | DatumVan            | 1935-01-01            |
+    | DatumTot            | 1936-01-01            |
     Dan worden de huwelijken/partners teruggegeven in de volgorde en met waarden:
-      | # | naam.voornamen | aangaanHuwelijkPartnerschap.jaar  | ontbindingHuwelijkPartnerschap.datum |
-      | 0 | Ayaan          | 1936                              |                                      |
+    | naam.voornamen | aangaanHuwelijkPartnerschap.datum | ontbindingHuwelijkPartnerschap.datum |
+    | Annechien      | 00000000                          |                                      |
 
-    Als de partnerhistorie wordt gevraagd van de persoon met burgerservicenummer 999991553 met datumTotEnMet "1936-01-01"
+Rule : Als een huwelijk/partnerschap, actueel of ontbonden, in onderzoek is, en dit onderzoek is niet afgerond (Datum einde onderzoek is leeg), wordt inOnderzoek gevuld voor betreffende huwelijk/partnerschap.
```

Comment:
Waarom is er een aparte regel opgenomen voor niet opnemen beëindigd onderzoek, terwijl verder niks gezegd wordt over hoe onderzoek moet worden gevuld in de proxy? Beide volgen uit de in_onderzoek.feature, die wellicht ook moet worden uitgewerkt voor historie.

Reply:
Ik snap deze opmerking niet zo goed. Deze rule gaat juist over het wel opnemen van een niet beëindigd onderzoek. Ik kan me overigens best voorstellen dat deze rule in een aparte in_onderzoek feature moet worden opgenomen.